src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let transState = { ...state };
+  const cloneState = [];
+
+  for (const i in actions) {
```

Comment:
Using `for...in` to iterate over arrays is not recommended as it iterates over all enumerable properties, not just the array elements. Consider using `for...of` or a standard `for` loop instead.